### DISCUSSION TITLE
What's the difference between maxConcurrency and test.maxThreads?

### DISCUSSION THREAD

**Discussion by gajus** [vitest-dev/vitest#4356] [Q&A] [answered]
Vitest v1.0.0 added option `maxConcurrency`.

What's the difference between this and `maxThreads`?

**Comment by sheremet-va** [comment, marked as answer]
https://vitest.dev/config/#maxconcurrency

`maxConcurrency` was added [long before 1.0](https://github.com/vitest-dev/vitest/releases/tag/v0.15.0). `maxConcurrency` only affects `test.concurrent`.

https://vitest.dev/config/#pooloptions-threads-maxthreads

`maxThreads` affects the number of workers Vitest spins up in parallel when running test files (one per test file)

**Reply by gajus** [reply]
Thanks. Looks like I was after `poolOptions.threads.maxThreads`.

I was thrown off when typing `max*` in the config gave `maxConcurrency` as a suggestion.
